source/module.c

```diff
@@ -1074,31 +1074,66 @@ static void s_install_crash_handler(void) {
 #endif
 }
 
+static void s_uninstall_crash_handler(void) {
+#if defined(_WIN32)
+    SetUnhandledExceptionFilter(NULL);
+#elif defined(AWS_HAVE_EXECINFO)
+    sigaction(SIGSEGV, NULL, NULL);
+    sigaction(SIGABRT, NULL, NULL);
+    sigaction(SIGILL, NULL, NULL);
+    sigaction(SIGBUS, NULL, NULL);
+#endif
+}
+
+/*
+ * Temporary hack to detect multi-init so we can throw an exception because we haven't figured out the right way
+ * to support it yet.  Better than a hard crash in native code.
+ */
+static struct aws_mutex s_module_lock = AWS_MUTEX_INIT;
+static uint32_t s_module_initialize_count = 0;
+
 static void s_napi_context_finalize(napi_env env, void *user_data, void *finalize_hint) {
     (void)env;
     (void)finalize_hint;
-    aws_client_bootstrap_release(s_default_client_bootstrap);
-    aws_host_resolver_release(s_default_host_resolver);
-    aws_event_loop_group_release(s_node_uv_elg);
 
-    aws_thread_join_all_managed();
+    aws_mutex_lock(&s_module_lock);
+    --s_module_initialize_count;
+
+    if (s_module_initialize_count == 0) {
+        aws_client_bootstrap_release(s_default_client_bootstrap);
+        s_default_client_bootstrap = NULL;
+
+        aws_host_resolver_release(s_default_host_resolver);
+        s_default_host_resolver = NULL;
+
+        aws_event_loop_group_release(s_node_uv_elg);
+        s_node_uv_elg = NULL;
 
-    aws_unregister_log_subject_info_list(&s_log_subject_list);
-    aws_unregister_error_info(&s_error_list);
-    aws_auth_library_clean_up();
-    aws_mqtt_library_clean_up();
+        aws_thread_join_all_managed();
+
+        s_uninstall_crash_handler();
```

Comment:
trivial: maybe uninstall the crash handler later on?

Reply:
> trivial: maybe uninstall the crash handler later on?

Originally I was worried that the backtrace usage would require init to have happened first (for example, if we called aws_backgrace_log) but since we use aws_backtrace_print and the impl is only hitting posix functions and one byte buffer usage, I think it's safe to make the crash handler wrap the library init/cleanup.